mule-extensions-api-persistence/src/main/java/org/mule/runtime/extension/internal/persistence/metadata/dto/EntityMetadata.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.persistence.metadata.dto;
+
+import org.mule.runtime.api.metadata.descriptor.ImmutableTypeMetadataDescriptor;
+import org.mule.runtime.api.metadata.descriptor.TypeMetadataDescriptor;
+import org.mule.runtime.api.metadata.resolving.MetadataResult;
+
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.mule.runtime.api.metadata.resolving.MetadataResult.failure;
+import static org.mule.runtime.api.metadata.resolving.MetadataResult.success;
+
+/**
+ * DTO that represents a {@link TypeMetadataDescriptor} into a serializable format.
```

Comment:
we can mention that this works with MetadataResult and that is related to the Entity of DSQL